.pre-commit-config.yaml

```diff
@@ -1,72 +1,81 @@
+exclude: |
+          (?x)(
+              ^tests/.*/fixtures/.*
+              | ^src/poetry/core/_vendor
+          )
+
 repos:
-  - repo: https://github.com/psf/black
-    rev: 21.9b0
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
     hooks:
-      - id: black
-        exclude: ^src/poetry/core/_vendor
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: debug-statements
+      - id: check-merge-conflict
+      - id: check-case-conflict
+      - id: check-json
+      - id: check-toml
+      - id: check-yaml
+      - id: pretty-format-json
+        args:
+          - --autofix
+          - --no-ensure-ascii
+          - --no-sort-keys
+      - id: check-ast
+      - id: debug-statements
+      - id: check-docstring-first
 
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.9.2
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
     hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-annotations
-          - flake8-bugbear
-          - flake8-comprehensions
-          - flake8-eradicate
-          - flake8-simplify
-          - flake8-tidy-imports
-          - flake8-use-fstring
+      - id: python-check-mock-methods
+      - id: python-use-type-annotations
+      - id: python-check-blanket-type-ignore
+      - id: python-check-blanket-noqa
 
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.910
+  - repo: https://github.com/asottile/yesqa
+    rev: v1.3.0
     hooks:
-      - id: mypy
-        pass_filenames: false
+      - id: yesqa
+        additional_dependencies: &flake8_deps
+          - flake8-annotations==2.7.0
+          - flake8-bugbear==22.1.11
+          - flake8-comprehensions==3.8.0
+          - flake8-eradicate==1.2.0
+          - flake8-simplify==0.15.1
+          - flake8-tidy-imports==4.6.0
+          - flake8-use-fstring==1.3
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py36-plus
 
   - repo: https://github.com/pycqa/isort
-    rev: 5.9.3
+    rev: 5.10.1
     hooks:
       - id: isort
-        additional_dependencies: [toml]
         exclude: |
           (?x)(
               ^.*/?setup\.py$
-              | ^src/poetry/core/_vendor
               | tests/.*\.pyi$
           )
 
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v2.29.1
+  - repo: https://github.com/psf/black
+    rev: 21.12b0
     hooks:
-      - id: pyupgrade
-        args:
-          - --py36-plus
-        exclude: ^src/poetry/core/_vendor
+      - id: black
 
-  - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.0.1
+  - repo: https://gitlab.com/pycqa/flake8
```

Comment:
In poetry, the repo from github is used. Probably, there is no reason to use gitlab, here.

Reply:
:+1: